### ISSUE TITLE
RFE: Allow paths to reconstruct and quota to be configurable somehow

### ISSUE THREAD

**Issue by jasontibbitts** [cyrusimap/cyrus-imapd#2386]
While tracking down why a Cassandane test that calls the IMAP RECONSTRUCT command fails, I found that two executables called by imapd are hardcoded to be found in SBIN_DIR and nowhere else: reconstruct and quota.  There's a third instance where ctl_deliver -E is implemented to print a warning and call SBIN_DIRT/cyr_expire, but I doubt that's a real problem and it can probably be removed at some point anyway.

One problem with hardcoding the paths to reconstruct and quota is is that you can't completely run an imapd instance outside of the location it was compiled to run in.  Which obviously isn't a huge deal, but does prevent certain types of testing like the the Fedora package does (where it's compiled to live in the usual system location but gets installed with make install DESTDIR=random_path and then tested from there).  Basically all of Cassandane works fine this way with an appropriate configuration, except for one test that calls reconstruct.

Also, looking at this made me realize that Fedora renames quota to cyr_quota but doesn't patch the call to it.  I'll have to fix that.  Being able to test it via Cassandane would have revealed that problem a long time ago.

**Comment by elliefm** [comment]
Oh heh, reading my notifications in chronological order.  FTR here's my comment from #2047:

> Oooh that's a curly one
> 
> We can't detect a destdir setting at compile time, because it's an install-time override. The "compile it as if it will be installed in /prefix, but actually install it to /destdir/prefix" behaviour is literally the whole point! I imagine it would do the right thing if it was running chroot'd to /destdir, but that's a whole nother can of worms
> 
> Off the top of my head, I think the way to handle this would probably be to add optional imapd.conf settings to override the locations exec'd by imapd. Ideally this would take the form of auditing all the things that get exec'd and doing something for all of them (vs tackling them piecemeal and winding up with a mess). There'd be security aspects to take into consideration as well, I guess

**Comment by jasontibbitts** [comment]
Yep, just an option to set a path (which defaults to SBIN_DIR) where these executables live would do it.  I don't think they need to be individually configurable.

**Comment by brong** [comment]
Just closing this, I think it's fine - we are planning to go more containerised.  Maybe for Cassandane it's tricky but meh, feel free to re-open if it's still a problem.
